src/common/components/HashGenerator/index.js

```diff
@@ -0,0 +1,15 @@
+function HashGenerator(hash) {
+  const hashLength = hash.toString().length;
+
+  if (hashLength === 1) {
+    return `000${hash}`;
+  } else if (hashLength === 2) {
+    return `00${hash}`;
+  } else if (hashLength === 3) {
+    return `0${hash}`;
+  } else {
+    return hash;
+  }
+}
+
+export default HashGenerator;
```

Comment:
String.prototype.padStart() can be used instead
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart